andela/ah-jumanji
# #161967012 Enable Social login via Google, Facebook and Twitter

#### What does this PR do?
Adds social login functions to the application
#### Description of Task to be completed?
Users should be able to login to the application using their social platforms. 
- `Facebook`
- `Twitter`
- `Google`

#### How should this be manually tested?
1. Git clone the repo using this command:
  `https://github.com/andela/ah-jumanji.git`

2. Install all packages from requirements.txt
  `pip install -r requirements.txt`

3. Create migrations and migrate 
   `python manage.py makemigrations` && `python manage.py migrate`

4. Run the application
  `python manage.py runserver`

5. Access the page through: 
   `http://127.0.0.1:8000`

6. Ensure all `.env` variables are filled appropriately
```
# Twitter oauth properties 
SOCIAL_AUTH_TWITTER_KEY='<twitter-app-key>'
SOCIAL_AUTH_TWITTER_SECRET='<twitter-app-secret>'

# Facebook oauth properties
SOCIAL_AUTH_FACEBOOK_KEY='<facebook-app-key>'
SOCIAL_AUTH_FACEBOOK_SECRET=''<facebook-app-secret>'
SOCIAL_AUTH_FACEBOOK_API_VERSION='<facebook-api-version>'

# Google oauth properties
SOCIAL_AUTH_GOOGLE_OAUTH2_KEY='<google-oauth2-app-key>'
SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET='<google-oauth2-app-secret>'
```
#### Any background context you want to provide?
Social login is essential in the new technologies as it thelps in safeguarding users details and assists the platform in validation of the users. Plus the degree of safety in user details management can be done easily.
#### What are the relevant pivotal tracker stories?
#[161967012](https://www.pivotaltracker.com/n/projects/2225604)

#### Screenshots (if appropriate)
Sample Twtter Request

<img width="974" alt="screenshot 2018-12-21 at 10 29 07" src="https://user-images.githubusercontent.com/41139653/50330252-8ecba280-050b-11e9-9bda-00a697eb3b58.png">

Sample request

<img width="978" alt="screenshot 2018-12-21 at 10 29 35" src="https://user-images.githubusercontent.com/41139653/50330304-c1759b00-050b-11e9-9b2b-b48b9b3fc81a.png">

#### Questions:

#### Checklist:
